clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -36,13 +36,26 @@ void processRecordDecl(RecordDecl *Declaration, ProgramInfo &Info,
         // We only want to re-write a record if it contains
         // any pointer types, to include array types.
         for (const auto &D : Definition->fields()) {
-          if (D->getType()->isPointerType() || D->getType()->isArrayType()) {
-            if(FL.isInSystemHeader() || Definition->isUnion()) {
-              CVarSet C = Info.getVariable(D, Context);
-              std::string Rsn = "External struct field or union encountered";
-              CB.constraintAllCVarsToWild(C, Rsn, nullptr);
+          bool mark_wild = ((D->getType()->isPointerType()
+                             || D->getType()->isArrayType()) &&
+                            (FL.isInSystemHeader() || Definition->isUnion()));
```

Comment:
Can you add a comment above your definition of `mark_wild` to explain what all of this means?